AGENTS.md

```diff
@@ -0,0 +1,433 @@
+### Naming
+
+- **Name**: Must be a valid Julia identifier. Replace spaces/dashes with underscores. Match upstream casing.
+- **Version**: Only `X.Y.Z` format. Truncate any `-alpha`, `+build`, or 4+ level versions.
+- **Products**: Export symbols match the library/executable names (as symbols: `:libname`).
+
+### Sources
+
+- **ArchiveSource**: For tarballs (`.tar.gz`, `.tar.xz`, `.zip`). Always include SHA256 hash.
+- **GitSource**: For git repos. Use specific commit hash, not branch names.
+- **DirectorySource**: For local patches. Place patches in `bundled/patches/` subdirectory.
+- Build **one package per recipe**. Don't bundle multiple packages—use dependencies instead.
+
+### Build Script
+
+The script runs in `x86_64-linux-musl` environment. Key variables:
+
+- `${prefix}`: Install root (target for all outputs)
+- `${bindir}`: Executables go here (= `${prefix}/bin`)
+- `${libdir}`: Libraries go here (= `${prefix}/bin` on Windows, `${prefix}/lib` elsewhere)
+- `${includedir}`: Headers go here (= `${prefix}/include`)
+- `${WORKSPACE}/srcdir`: Where sources are extracted
+- `${target}`: Target platform triplet
+- `${nproc}`: Number of parallel jobs
+- `${CC}`, `${CXX}`, `${FC}`: Cross-compilers for C/C++/Fortran
+
+### Common Build Systems
+
+**Autotools**:
+
+```bash
+./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target}
+make -j${nproc}
+make install
+```
+
+Use `update_configure_scripts` first if `config.sub` is old.
+
+**CMake**:
+
+```bash
+cmake -B build \
+    -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+    -DCMAKE_BUILD_TYPE=Release
+cmake --build build --parallel ${nproc}
+cmake --install build
+```
+
+**Meson**:
+
+```bash
+meson setup build --cross-file="${MESON_TARGET_TOOLCHAIN}"
+meson compile -C build -j${nproc}
+meson install -C build
+```
+
+**Make**:
+
+```bash
+make -j${nproc} PREFIX=${prefix}
+make install PREFIX=${prefix}
+```
+
+**Go** (add `compilers=[:c, :go]`):
+
+```bash
+go build -o ${bindir}/executable
+```
+
+**Rust** (add `compilers=[:c, :rust]`):
+
+```bash
+cargo build --release
+install -Dm755 target/*/release/executable ${bindir}/executable
+```
+
+### Platform-Specific Logic
+
+```bash
+if [[ "${target}" == *-mingw* ]]; then
+    # Windows-specific
+elif [[ "${target}" == *-apple-* ]]; then
+    # macOS-specific
+elif [[ "${target}" == *-freebsd* ]]; then
+    # FreeBSD-specific
+fi
+
+if [[ ${nbits} == 32 ]]; then
+    # 32-bit specific
+fi
+```
+
+### Products
+
+- **LibraryProduct**: Shared libraries (`.so`, `.dylib`, `.dll`)
+- **ExecutableProduct**: Binary executables
+- **FileProduct**: Other files (headers, data files)
+- **FrameworkProduct**: macOS frameworks
+
+### Dependencies
+
+- **Dependency**: Runtime dependency (will be in JLL package)
+- **BuildDependency**: Build-time only (not in final JLL)
+- **HostBuildDependency**: Runs on build host, not target
+
+Always add `_jll` suffix: `Dependency("Zlib_jll")`
+
+### GCC Version Selection
```

Comment:
Can we link other documents here? It needs to be one of the versions we actually have built, so basically only things in this table: https://github.com/JuliaPackaging/Yggdrasil/blob/master/RootFS.md#compiler-shards